python/cudf/cudf/core/column/string.py

```diff
@@ -117,10 +141,17 @@ def _return_or_inplace(self, new_col, **kwargs):
                 # This branch indicates the passed as new_col
                 # is actually a table-like data
                 table = new_col
-                return self._parent._constructor_expanddim(
-                    {index: value for index, value in enumerate(table)},
-                    index=self._parent.index,
-                )
+                from cudf._libxx.table import Table
+
+                if isinstance(table, Table):
+                    return self._parent._constructor_expanddim(
+                        data=table._data
+                    )
+                else:
+                    return self._parent._constructor_expanddim(
+                        {index: value for index, value in enumerate(table)},
+                        index=self._parent.index,
+                    )
```

Comment:
Can you explain this change?

Reply:
So In case of few apis the `new_col`/`table` is a `Table` object which doesn't support iteration and thus fails at `enumerate`, hence added a special check for objects of type `Table`